ios/MullvadVPN/TunnelManager.swift

```diff
@@ -677,17 +684,20 @@ class TunnelManager {
         self.tunnelProvider = tunnelProvider
 
         // Set up tunnel IPC
-        if let session = connection as? NETunnelProviderSession {
+        if let session = connection as? VPNTunnelProviderSessionProtocol {
             self.tunnelIpc = PacketTunnelIpc(session: session)
         }
 
         // Register for tunnel connection status changes
         tunnelStatusSubscriber = NotificationCenter.default.publisher(for: .NEVPNStatusDidChange, object: connection)
-            .receive(on: executionQueue)
-            .sink { [weak self] (notification) in
-                guard let connection = notification.object as? NEVPNConnection else { return }
+            .map { notification in
```

Comment:
Avoids data race. Accessing `connection.status` from a different thread may not return the same `status` that was set before the notification was received. Copying the `status` synchronously in response to the notification is on the other side guarantees that the `status` is consistent, then we can dispatch it on the other thread.